crates/primitives/src/block.rs

```diff
@@ -180,15 +184,12 @@ impl Serialize for BlockId {
         S: Serializer,
     {
         match *self {
-            BlockId::Hash(ref x) => {
-                let mut s = serializer.serialize_struct("BlockIdEip1898", 1)?;
-                s.serialize_field("blockHash", &format!("{x:?}"))?;
-                s.end()
-            }
-            BlockId::HashReqCanonical((ref x, ref c)) => {
+            BlockId::Hash(BlockHashCanonical { ref block_hash, ref require_canonical }) => {
                 let mut s = serializer.serialize_struct("BlockIdEip1898", 1)?;
-                s.serialize_field("blockHash", &format!("{x:?}"))?;
-                s.serialize_field("requireCanonical", &format!("{c:?}"))?;
+                s.serialize_field("blockHash", &format!("{block_hash:?}"))?;
```

Comment:
block_hash should be `Serialize`
```suggestion
                s.serialize_field("blockHash", block_hash))?;
```